### ISSUE TITLE
Ternary operator used inside tuple creation expression causes compiler errors

### ISSUE THREAD

**Issue by 0xd4d** [dotnet/roslyn#17135]
**Version Used**: VS2017RC.4

```C#
using System;

namespace N {
	class Program {
		static object Field1;
		static object Field2;

		public void Method(bool b) {
			// Works (fails if you uncomment last statement)
			//(string, object) vt1 = new ValueTuple<string, object>("", b ? Field2 : Field1);

			// Works (fails if you uncomment last statement)
			//(string, object) vt2 = ValueTuple.Create("", b ? Field2 : Field1);

			// Works (always)
			//(string, object) vt3 = ("", b ? "" : (object)123);

			// Fails
			(string, object) vt4 = ("", b ? Field2 : Field1);
		}

		public static void Main() {
		}
	}
}
```

```
1>X:\ConsoleApp34\ConsoleApp34\Program.cs(19,43,19,44): error CS1026: ) expected
1>X:\ConsoleApp34\ConsoleApp34\Program.cs(19,43,19,44): error CS1002: ; expected
1>X:\ConsoleApp34\ConsoleApp34\Program.cs(19,43,19,44): error CS1513: } expected
1>X:\ConsoleApp34\ConsoleApp34\Program.cs(19,51,19,52): error CS1002: ; expected
1>X:\ConsoleApp34\ConsoleApp34\Program.cs(19,51,19,52): error CS1513: } expected
```

When hovering over the error (span: `b ? Field2`), QuickInfo shows two errors:

- A declaration is not allowed in this context.
- Use of unassigned local variable 'Field2'

Is this correct behavior? Did I miss something obvious?

**Comment by controlflow** [comment]
Looks like this is caused by allowed declaration expression parsing inside tuple literal expression.
`b? Field` is parsed as a declaration expression of nullable type `b` with `Field` variable designation, `:` is unexpected token after...

**Comment by gafter** [comment]
Fixed in #16834.
